README.md
# NAME

Dist::Zilla::PluginBundle::MAXMIND - MAXMIND's plugin bundle

# VERSION

version 0.84

# SYNOPSIS

    name    = My-Module
    author  = Dave Rolsky <autarch@urth.org>
    license = Artistic_2_0
    copyright_holder = Dave Rolsky

    [@MAXMIND]
    dist = My-Module
    ; Fefault is MakeMaker - or set it to ModuleBuild
    make_tool = MakeMaker
    ; These files won't be added to tarball
    exclude_files = ...
    ; Default is MAXMIND
    authority = MAXMIND
    ; Passed to AutoPrereqs - can be repeated
    prereqs_skip = ...
    ; Passed to Test::Pod::Coverage::Configurable if set
    pod_coverage_also_private = ...
    ; Passed to Test::Pod::Coverage::Configurable if set - can be repeated
    pod_coverage_class = ...
    ; Passed to Test::Pod::Coverage::Configurable if set - can be repeated
    pod_coverage_skip = ...
    ; Passed to Test::Pod::Coverage::Configurable if set - can be repeated
    pod_coverage_trustme = ...
    ; For pod spelling test - can be repeated
    stopwords = ...
    ; Can also put them in a separate file
    stopwords_file = ..
    ; Defaults to false
    use_github_homepage = 0
    ; Defaults to false
    use_github_issues = 0

# DESCRIPTION

This is the [Dist::Zilla](https://metacpan.org/pod/Dist%3A%3AZilla) plugin bundle I use for my distributions. Don't use
this directly for your own distributions, but you may find it useful as a
source of ideas for building your own bundle.

This bundle uses [Dist::Zilla::Role::PluginBundle::PluginRemover](https://metacpan.org/pod/Dist%3A%3AZilla%3A%3ARole%3A%3APluginBundle%3A%3APluginRemover) and
[Dist::Zilla::Role::PluginBundle::Config::Slicer](https://metacpan.org/pod/Dist%3A%3AZilla%3A%3ARole%3A%3APluginBundle%3A%3AConfig%3A%3ASlicer) so I can remove or
configure any plugin as needed.

This is more or less equivalent to the following `dist.ini`:

    ; Picks one of these - defaults to MakeMaker
    [MakeMaker]
    [ModuleBuild]

    [Git::GatherDir]
    ; Both are configured by setting exclude_files for the bundle. Simple
    ; values like "./foo/bar.pl" are treated as filenames, others like
    ; "*\.jnk$" are treated as a regex.
    exclude_filenames = ...
    exclude_match     = ...

    [ManifestSkip]
    [License]
    [ExecDir]
    [ShareDir]
    [Manifest]
    [CheckVersionIncrement]
    [TestRelease]
    [ConfirmRelease]
    [UploadToCPAN]
    ; Opens up the main module and finds a $VERSION
    [MAXMIND::VersionProvider]

    [Authority]
    ; Configured by setting authority for the bundle
    authority  = ...
    do_munging = 0

    [AutoPrereqs]
    ; Configured by setting skip_prereqs for the bundle
    skip = ...

    [CopyFilesFromBuild]
    copy = Build.PL
    copy = CONTRIBUTING.md
    copy = LICENSE
    copy = Makefile.PL
    copy = README.md
    copy = cpanfile
    copy = ppport.h

    [GitHub::Meta]
    ; Configured by setting use_github_issues for the bundle
    bugs = 0
    ; Configured by setting use_github_homepage for the bundle
    homepage = 0

    [MetaResources]
    homepage = http://metacpan.org/release/My-Module
    ; RT bits are omitted if use_github_issue is true
    bugtracker.web  = http://rt.cpan.org/Public/Dist/Display.html?Name=My-Module
    bugtracker.mail = bug-My-Module@rt.cpan.org

    [MetaProvides::Pckage]
    meta_noindex = 1

    [Meta::Contributors]
    [Meta::Config]
    [MetaJSON]
    [MetaYAML]

    [NextRelease]
    ; Width is configured by setting next_release_width for the bundle
    format = %-8v %{yyyy-MM-dd}d%{ (TRIAL RELEASE)}T

    ; Scans the test files for use of Test2 and picks either
    [Prereqs / Test::More with Test2]
    -phase = test
    -type  = requires
    Test::More = 1.302015

    ; If the distro doesn't use Test2
    [Prereqs / Test::More with subtest]
    -phase = test
    -type  = requires
    Test::More = 0.96

    [Prereqs / Modules for use with tidyall]
    -phase = develop
    -type  = requires
    Code::TidyAll::Plugin::Test::Vars = 0.02
    Parallel::ForkManager'            = 1.19
    Perl::Critic                      = 1.126
    Perl::Tidy                        = 20160302
    Test::Vars                        = 0.009

    [Prereqs / Test::Version which fixes https://github.com/plicease/Test-Version/issues/7]
    -phase = develop
    -type  = requires
    Test::Version = 2.05

    [PromptIfStale]
    phase  = build
    module = Dist::Zilla::PluginBundle::MAXMIND

    [PromptIfStale]
    phase = release
    check_all_plugins = 1
    check_all_prereqs = 1
    check_authordeps  = 1
    skip = Dist::Zilla::Plugin::MAXMIND::CheckChangesHasContent
    skip = Dist::Zilla::Plugin::MAXMIND::Contributors
    skip = Dist::Zilla::Plugin::MAXMIND::License
    skip = Dist::Zilla::Plugin::MAXMIND::TidyAll
    skip = Dist::Zilla::Plugin::MAXMIND::VersionProvider
    skip = Pod::Weaver::PluginBundle::MAXMIND

    [Test::Pod::Coverage::Configurable]
    ; Configured by setting pod_coverage_class for the bundle
    class = ...
    ; Configured by setting pod_coverage_skip for the bundle
    skip = ...
    ; Configured by setting pod_coverage_trustme for the bundle
    trustme = ...

    [Test::PodSpelling]
    ; Configured by setting stopwords and/or stopwords_file for the bundle
    stopwods = ...

    [PodSyntaxTests]

    [RunExtraTests]
    [MojibakeTests]
    [Test::CleanNamespaces]
    [Test::CPAN::Changes]
    [Test::CPAN::Meta::JSON]
    [Test::EOL]
    [Test::NoTabs]
    [Test::Portability]
    [Test::Synopsis]

    [Test::TidyAll]
    verbose = 1
    jobs    = 4
    minimum_perl = 5.010

    [Test::Compile]
    xt_mode = 1

    [Test::ReportPrereqs]
    verify_prereqs = 1

    [Test::Version]
    is_strict = 1

    ; Generates/updates a .mailmap file
    [MAXMIND::Contributors]
    [Git::Contributors]

    [SurgicalPodWeaver]
    ; See Pod::Weaver::PluginBundle::MAXMIND in this same distro for more info
    config_plugin = @MAXMIND

    ; Nasty hack so I can pass config from the dist.ini to the Pod::Weaver
    ; bundle. Currently used so I can set
    ; "MAXMIND::WeaverConfig.include_donations_pod = 0" in a dist.ini file.
    [MAXMIND::WeaverConfig]

    [ReadmeAnyFromPod / README.md in build]
    type     = markdown
    filename = README.md
    location = build
    phase    = build

    [GenerateFile::FromShareDir / Generate CONTRIBUTING.md]
    -dist     = Dist-Zilla-PluginBundle-MAXMIND
    -filename = CONTRIBUTING.md
    ; This is determined by looking through the distro for .xs files.
    has_xs    = ...

    [InstallGuide]
    [CPANFile]

    ; Only added if the distro has .xs files
    [PPPort]

    ; Like the default License plugin except that it defaults to Artistic 2.0.
    ; Also, if the copyright_year for the bundle is not this year, it passes
    ; something like "2014-2016" to Software::License.
    [MAXMIND::License]

    [CheckPrereqsIndexed]

    ; More or less like Dist::Zilla::Plugin::CheckChangesHasContent but uses
    ; CPAN::Changes to parse the Changes file.
    [MAXMIND::CheckChangesHasContent]

    [Git::CheckFor::MergeConflicts]

    ; Generates/updates tidyall.ini, perlcriticrc, and perltidyrc
    [MAXMIND::TidyAll]

    ; The allow_dirty list is basically all of the generated or munged files
    ; in the distro, including:
    ;     Build.PL
    ;     CONTRIBUTING.md
    ;     Changes
    ;     LICENSE
    ;     Makefile.PL
    ;     README.md
    ;     cpanfile
    ;     ppport.h
    ;     tidyall.ini
    [Git::Check]
    allow_dirty = ...

    [Git::Commit / Commit generated files]
    allow_dirty = ...

    [Git::Tag]
    [Git::Push]

    [BumpVersionAfterRelease]

    [Git::Commit / Commit version bump]
    allow_dirty_match = .+
    commit_msg        = Bump version after release

    [Git::Push / Push version bump]

# SUPPORT

Bugs may be submitted through [https://github.com/maxmind/Dist-Zilla-PluginBundle-MAXMIND/issues](https://github.com/maxmind/Dist-Zilla-PluginBundle-MAXMIND/issues).

# AUTHOR

Dave Rolsky <autarch@urth.org>

# CONTRIBUTORS

- Dave Rolsky <drolsky@maxmind.com>
- Florian Ragwitz <rafl@debian.org>
- Greg Oschwald <goschwald@maxmind.com>
- Mark Fowler <mark@twoshortplanks.com>
- Olaf Alders <oalders@maxmind.com>
- Patrick Cronin <PatrickCronin@users.noreply.github.com>
- Patrick Cronin <pcronin@maxmind.com>

# COPYRIGHT AND LICENSE

This software is Copyright (c) 2021 by Dave Rolsky and MaxMind, Inc.

This is free software, licensed under:

    The Artistic License 2.0 (GPL Compatible)


Changes
{{$NEXT}}

0.84     2021-06-28

- Drop CorrectBranch plugin as we don't allow pushing to the main
  branch and want even release commits reviewed.
- Update perlcriticrc theme to only include core and moose policies.


0.83     2017-10-10

- Added missing Path::Tiny import.


0.82     2017-10-06

- Synced Perl::Tidy and Perl::Critic configuration files with internal
  files.


0.81     2017-07-17

- Updated perlcriticrc. Added Freenode::WhileDiamondDefaultAssignment.


0.80     2017-02-02

- Use ~/.pause-maxmind rather than ~/.pause.


0.79     2017-01-05

- Fork DROLSKY bundle to make a separate MAXMIND bundle.


0.78     2016-11-29

- Handle a ppport.h file being generated in a subdirectory as well as in the
  root of the distro.


0.77     2016-11-12

- Generated POD said my IRC nick was "drolsky" instead of "autarch". Reported
  by mst.


0.76     2016-11-01

- Only tell CopyFileFromBuild to copy files that which actually exist in the
  distro. The latest version of that plugin now dies on missing files.


0.75     2016-10-11

- Added pod_coverage_also_private to pass more stuff to
  Test::Pod::Coverage::Configurable


0.74     2016-10-07

- Added CheckSelfDependency and CheckStrictVersion plugins.


0.73     2016-10-02

- Set jobs for Test::TidyAll plugin to 4 by default.


0.72     2016-09-25

- Make the Pod::Weaver bits work when used from a subclass of this bundle.


0.71     2016-09-17

- Removed Module::Metadata prereq and require Test::Version 2.05 instead.

- Added more docs.


0.70     2016-09-16

- Explicitly add Module::Metadata 1.000033 to the develop prereqs in order to
  work around bugs with Test::Version and earlier versions of
  Module::Metadata. See https://github.com/plicease/Test-Version/issues/7 for
  the relevant bug report.


0.69     2016-09-13

- Very tiny wording change in CONTRIBUTING.md.


0.68     2016-09-12

- Require Perl::Critic::Moose 1.05;


0.67     2016-08-20

- Various tweaks and cleanups for the pod weaver bundle.


0.66     2016-08-13

- Implemented a rather convoluted way of passing config into my Pod::Weaver
  bundle in order to allow for disabling the DONATIONS pod section on a
  per-distro basis.


0.65     2016-07-04

- Avoid warnings when t/ directory contains non-UTF8 data.


0.64     2016-06-26

- If the distro uses Test2 modules, the Test::More prereq is now set to
  1.302015 instead of 0.96. Suggested by Graham Knop.


0.63     2016-06-17

- Small tweak to generated perlcriticrc.


0.62     2016-06-12

- Only run PromptIfStale for release phase.


0.61     2016-06-05

- Don't include GitHub::Meta when running under Travis.


0.60     2016-06-04

- Don't include PromptIfStale when running under Travis. It's not like I can
  stop the build and update prereqs anyway.


0.59     2016-05-30

- Make the generated tidyall.ini only use Test::Vars on modules. It ends up
  running code that it checks, so if you run it on .t files it actually runs
  them, causing all sorts of problems with the tidyall tests.


0.58     2016-05-29

- Set the minimum_perl for the Test::Tidyall plugin to 5.10 since Test::Vars
  requires that version.


0.57     2016-05-28

- Move stale module check to build phase.


0.56     2016-05-21

- Add the Test::Vars tidyall plugin to my default tidyall.ini.


0.55     2016-05-18

- Fix handling of regexes in "exclude" config values.


0.54     2016-05-18

- Fix check for XS to check outside of the top-level dir.


0.53     2016-05-08

- Add psgi to program-extensions in perlcriticrc.


0.52     2016-05-06

- Set program-extensions in perlcriticrc.


0.51     2016-05-06

- Make sure we set argv for all PerlCritic & PerlTidy sections when munging
  tidyall.ini.


0.50     2016-04-25

- Always make the Test::TidyAll plugin verbose. There's no downside to seeing
  diffs on tidy failures.


0.49     2016-04-24

- When adding Perl::Critic and Perl::Tidy to development prereqs, ask for the
  latest versions.


0.48     2016-04-24

- Only include the DONATIONS pod section for distros where I am the copyright
  holder.


0.47     2016-04-11

- Revert changes in 0.46 that required a recent Perl. This makes testing on
  Travis painful and makes it harder for others to contribute to any module of
  mine which uses this bundle.


0.46     2016-04-11

- Switched from [ExtraTests] to [RunExtraTests].

- Made the TidyAll plugin in this bundle rewrite existing tidyall.ini files to
  make sure we don't try to check or tidy generated xt tests.


0.45     2016-03-20

- Make sure generated Pod does not have trailing whitespace.

- Fix links to mailing lists to be "mailto:" links.


0.44     2016-03-20

- Make sure that all the Pod::Weaver-related modules that this distribution
  uses are added to the prereqs list.


0.43     2016-03-20

- Added [CheckChangesHasContent].

- Added [Test::CLeanNamespaces].

- If the distro has XS code [PPPort] is added and the generated ppport.h is
  copied from the build into the repo.

- This bundle now generates a CONTRIBUTING.md file automatically.

- Added a new Pod::Weaver author bundle that replaces the one-time generation
  of a weaver.ini file.

- Fixed README.md generation to happen after pod weaving, instead of before.


0.42     2016-02-01

- Disabled the Subroutines::ProhibitExplicitReturnUndef policy in the
  generated perlcriticrc file.


0.41     2016-01-29

- Added the Test::CPAN::Meta::JSON plugin.


0.40 2015-11-24

- Added a new plugin, MAXMIND::Git::CheckFor::CorrectBranch. This plugin only
  checks that the branch is correct on non-TRIAL releases.


0.39     2015-11-24

- Added a use_github_homepage config option.


0.38     2015-09-28

- Remove the RewriteVersion plugin and use my own version provider that simply
  reads the version from the main module. I don't want things appending '#
  TRIAL' comments to my version lines and un-tidying my code.


0.37     2015-08-15

- Include MojibakeTests plugin.


0.36     2015-07-12


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.022.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "MAXMIND's plugin bundle",
  "AUTHOR" => "Dave Rolsky <autarch\@urth.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-MAXMIND",
  "LICENSE" => "artistic_2",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "Dist::Zilla::PluginBundle::MAXMIND",
  "PREREQ_PM" => {
    "CPAN::Changes" => 0,
    "Code::TidyAll::Config::INI::Reader" => "0.44",
    "Dist::Zilla" => "6.0",
    "Dist::Zilla::Plugin::Authority" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::BumpVersionAfterRelease" => 0,
    "Dist::Zilla::Plugin::CPANFile" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
    "Dist::Zilla::Plugin::CheckSelfDependency" => 0,
    "Dist::Zilla::Plugin::CheckStrictVersion" => 0,
    "Dist::Zilla::Plugin::CheckVersionIncrement" => 0,
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::GenerateFile::FromShareDir" => "0.013",
    "Dist::Zilla::Plugin::Git::Check" => 0,
    "Dist::Zilla::Plugin::Git::CheckFor::MergeConflicts" => 0,
    "Dist::Zilla::Plugin::Git::Commit" => 0,
    "Dist::Zilla::Plugin::Git::Contributors" => 0,
    "Dist::Zilla::Plugin::Git::GatherDir" => 0,
    "Dist::Zilla::Plugin::Git::Push" => 0,
    "Dist::Zilla::Plugin::Git::Tag" => 0,
    "Dist::Zilla::Plugin::GitHub::Meta" => 0,
    "Dist::Zilla::Plugin::GitHub::Update" => 0,
    "Dist::Zilla::Plugin::InstallGuide" => 0,
    "Dist::Zilla::Plugin::Meta::Contributors" => 0,
    "Dist::Zilla::Plugin::MetaConfig" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
    "Dist::Zilla::Plugin::MetaResources" => 0,
    "Dist::Zilla::Plugin::MojibakeTests" => 0,
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::PPPort" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::PromptIfStale" => "0.050",
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
    "Dist::Zilla::Plugin::RunExtraTests" => 0,
    "Dist::Zilla::Plugin::SurgicalPodWeaver" => 0,
    "Dist::Zilla::Plugin::Test::CPAN::Changes" => 0,
    "Dist::Zilla::Plugin::Test::CPAN::Meta::JSON" => 0,
    "Dist::Zilla::Plugin::Test::CleanNamespaces" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::EOL" => "0.14",
    "Dist::Zilla::Plugin::Test::NoTabs" => 0,
    "Dist::Zilla::Plugin::Test::Pod::Coverage::Configurable" => 0,
    "Dist::Zilla::Plugin::Test::PodSpelling" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
    "Dist::Zilla::Plugin::Test::Synopsis" => 0,
    "Dist::Zilla::Plugin::Test::TidyAll" => "0.04",
    "Dist::Zilla::Plugin::Test::Version" => 0,
    "Dist::Zilla::Role::BeforeBuild" => 0,
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Role::LicenseProvider" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Dist::Zilla::Role::VersionProvider" => 0,
    "List::Util" => "1.45",
    "Module::Runtime" => 0,
    "Moose" => 0,
    "PadWalker" => 0,
    "Parse::PMFile" => 0,
    "Path::Iterator::Rule" => 0,
    "Path::Tiny" => 0,
    "Perl::Critic::Freenode" => "0.021",
    "Perl::Critic::Moose" => "1.05",
    "Pod::Elemental::Transformer::List" => 0,
    "Pod::Elemental::Transformer::Verbatim" => 0,
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::SingleEncoding" => 0,
    "Pod::Weaver::Plugin::Transformer" => 0,
    "Pod::Weaver::Section::AllowOverride" => 0,
    "Pod::Weaver::Section::Authors" => 0,
    "Pod::Weaver::Section::Collect" => 0,
    "Pod::Weaver::Section::Contributors" => 0,
    "Pod::Weaver::Section::GenerateSection" => 0,
    "Pod::Weaver::Section::Generic" => 0,
    "Pod::Weaver::Section::Leftovers" => 0,
    "Pod::Weaver::Section::Legal" => 0,
    "Pod::Weaver::Section::Name" => 0,
    "Pod::Weaver::Section::Region" => 0,
    "Pod::Weaver::Section::Version" => 0,
    "Sort::ByExample" => 0,
    "String::RewritePrefix" => 0,
    "autodie" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.85",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "CPAN::Changes" => 0,
  "Code::TidyAll::Config::INI::Reader" => "0.44",
  "Dist::Zilla" => "6.0",
  "Dist::Zilla::Plugin::Authority" => 0,
  "Dist::Zilla::Plugin::AutoPrereqs" => 0,
  "Dist::Zilla::Plugin::BumpVersionAfterRelease" => 0,
  "Dist::Zilla::Plugin::CPANFile" => 0,
  "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
  "Dist::Zilla::Plugin::CheckSelfDependency" => 0,
  "Dist::Zilla::Plugin::CheckStrictVersion" => 0,
  "Dist::Zilla::Plugin::CheckVersionIncrement" => 0,
  "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
  "Dist::Zilla::Plugin::GenerateFile::FromShareDir" => "0.013",
  "Dist::Zilla::Plugin::Git::Check" => 0,
  "Dist::Zilla::Plugin::Git::CheckFor::MergeConflicts" => 0,
  "Dist::Zilla::Plugin::Git::Commit" => 0,
  "Dist::Zilla::Plugin::Git::Contributors" => 0,
  "Dist::Zilla::Plugin::Git::GatherDir" => 0,
  "Dist::Zilla::Plugin::Git::Push" => 0,
  "Dist::Zilla::Plugin::Git::Tag" => 0,
  "Dist::Zilla::Plugin::GitHub::Meta" => 0,
  "Dist::Zilla::Plugin::GitHub::Update" => 0,
  "Dist::Zilla::Plugin::InstallGuide" => 0,
  "Dist::Zilla::Plugin::Meta::Contributors" => 0,
  "Dist::Zilla::Plugin::MetaConfig" => 0,
  "Dist::Zilla::Plugin::MetaJSON" => 0,
  "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
  "Dist::Zilla::Plugin::MetaResources" => 0,
  "Dist::Zilla::Plugin::MojibakeTests" => 0,
  "Dist::Zilla::Plugin::NextRelease" => 0,
  "Dist::Zilla::Plugin::PPPort" => 0,
  "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::PromptIfStale" => "0.050",
  "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
  "Dist::Zilla::Plugin::RunExtraTests" => 0,
  "Dist::Zilla::Plugin::SurgicalPodWeaver" => 0,
  "Dist::Zilla::Plugin::Test::CPAN::Changes" => 0,
  "Dist::Zilla::Plugin::Test::CPAN::Meta::JSON" => 0,
  "Dist::Zilla::Plugin::Test::CleanNamespaces" => 0,
  "Dist::Zilla::Plugin::Test::Compile" => 0,
  "Dist::Zilla::Plugin::Test::EOL" => "0.14",
  "Dist::Zilla::Plugin::Test::NoTabs" => 0,
  "Dist::Zilla::Plugin::Test::Pod::Coverage::Configurable" => 0,
  "Dist::Zilla::Plugin::Test::PodSpelling" => 0,
  "Dist::Zilla::Plugin::Test::Portability" => 0,
  "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
  "Dist::Zilla::Plugin::Test::Synopsis" => 0,
  "Dist::Zilla::Plugin::Test::TidyAll" => "0.04",
  "Dist::Zilla::Plugin::Test::Version" => 0,
  "Dist::Zilla::Role::BeforeBuild" => 0,
  "Dist::Zilla::Role::BeforeRelease" => 0,
  "Dist::Zilla::Role::LicenseProvider" => 0,
  "Dist::Zilla::Role::Plugin" => 0,
  "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "Dist::Zilla::Role::VersionProvider" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "List::Util" => "1.45",
  "Module::Runtime" => 0,
  "Moose" => 0,
  "PadWalker" => 0,
  "Parse::PMFile" => 0,
  "Path::Iterator::Rule" => 0,
  "Path::Tiny" => 0,
  "Perl::Critic::Freenode" => "0.021",
  "Perl::Critic::Moose" => "1.05",
  "Pod::Elemental::Transformer::List" => 0,
  "Pod::Elemental::Transformer::Verbatim" => 0,
  "Pod::Weaver::Config::Assembler" => 0,
  "Pod::Weaver::Plugin::SingleEncoding" => 0,
  "Pod::Weaver::Plugin::Transformer" => 0,
  "Pod::Weaver::Section::AllowOverride" => 0,
  "Pod::Weaver::Section::Authors" => 0,
  "Pod::Weaver::Section::Collect" => 0,
  "Pod::Weaver::Section::Contributors" => 0,
  "Pod::Weaver::Section::GenerateSection" => 0,
  "Pod::Weaver::Section::Generic" => 0,
  "Pod::Weaver::Section::Leftovers" => 0,
  "Pod::Weaver::Section::Legal" => 0,
  "Pod::Weaver::Section::Name" => 0,
  "Pod::Weaver::Section::Region" => 0,
  "Pod::Weaver::Section::Version" => 0,
  "Sort::ByExample" => 0,
  "String::RewritePrefix" => 0,
  "Test::More" => "0.96",
  "autodie" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}
